File: Bigger_is_Greater.cpp
#include <iostream>
#include <algorithm>
int main() {
    int k;
    std::cin>>k;
    for(int i=0;i<k;i++) {
        std::string input;
        std::cin>>input;
        if(std::next_permutation(input.begin(),input.end()))
            std::cout<<input<<std::endl;
        else
            std::cout<<"No Answer"<<std::endl;
    }
    return 0;
}

File: bigger_is_greater.py
for _ in range(int(input())):
    s = input()
    has_greater = False
    for i in range(len(s)-1)[::-1]:
        if ord(s[i]) < ord(s[i+1]):
            for j in range(i+1,len(s))[::-1]:
                if ord(s[i]) < ord(s[j]):
                    lis = list(s)
                    lis[i],lis[j]=lis[j],lis[i]
                    print("".join(lis[:i+1]+lis[i+1:][::-1]))
                    has_greater = True
                if has_greater == True:
                    break
        if has_greater == True:
            break
    if has_greater == False:
        print("no answer")
